questionReply/questionReply-library/src/main/java/org/silverpeas/components/questionreply/model/Reply.java

```diff
@@ -73,7 +83,36 @@ public String getCreatorId() {
     return creatorId;
   }
 
-  public String getCreationDate() {
+  @Override
+  public ContributionIdentifier getContributionId() {
+    return ContributionIdentifier.from(getPK());
+  }
+
+  @Override
+  public User getCreator() {
+    return User.getById(getCreatorId());
+  }
+
+  @Override
+  public Date getCreationDate() {
+    try {
+      return DateUtil.parse(this.getCreationDateAsString());
+    } catch (ParseException e) {
+      throw new RuntimeException(e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)